### DISCUSSION TITLE
how to use VirtualProgram to add texture2d with teccoords to an existing MPGeometry in application?

### DISCUSSION THREAD

**Discussion by 4129qq** [pelicanmapping/osgearth#1984] [Support Q&A]
MPGeometry* geom= dynamic_cast<MPGeometry*>( geode->getDrawable(0) );
osg::Texture2D* tex2d=new osg::Texture2D();
tex2d->setImage(..);
.....
osg::Vec2Array* coords=new osg::Vec2Array();
.....

**Comment by gwaldron** [comment]
Suggest using either (a) an `GDALImageLayer` or an `ImageOverlay` depending on your needs.

**Comment by 4129qq** [comment]
gwaldron：
&nbsp; &nbsp;hello!
&nbsp; &nbsp;https://github.com/gwaldron/osgearth/discussions/1988
&nbsp; &nbsp;I test in osgearth2.7&nbsp; how to solve?
&nbsp; &nbsp;thanks




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "gwaldron/osgearth"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月25日(星期五) 晚上9:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [gwaldron/osgearth] how to use VirtualProgram to add texture2d with teccoords to an existing MPGeometry in application? (Discussion #1984)





 
Suggest using either (a) an GDALImageLayer or an ImageOverlay depending on your needs.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
